src/Model/User.php

```diff
@@ -758,12 +758,16 @@ public static function getIdFromAuthenticateHooks(string $username, string $pass
 			'user_record'   => null
 		];
 
-		/*
+		$eventDispatcher = DI::eventDispatcher();
+
+		/**
 		 * An addon indicates successful login by setting 'authenticated' to non-zero value and returning a user record
 		 * Addons should never set 'authenticated' except to indicate success - as hooks may be chained
 		 * and later addons should not interfere with an earlier one that succeeded.
```

Comment:
isn't that a requirement for addon constraints/policies? just asking :)

Reply:
We can enforce this in the future with special StoppableEvents or `$event->setAsAuthenticated()`.